File: src/main/java/com/example/dpop/ext_stammdaten/internal/Person.java
package com.example.dpop.ext_stammdaten.internal;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "person")
public class Person {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "kvnr", unique = true, nullable = false, length = 20)
    private String kvnr;

    private String name;
    private String vorname;
    private String strasse;
    private String hausnummer;
    private String plz;
    private String ort;

    protected Person() {
    }

    public Person(String kvnr, String name, String vorname, String strasse, String hausnummer, String plz, String ort) {
        this.kvnr = kvnr;
        this.name = name;
        this.vorname = vorname;
        this.strasse = strasse;
        this.hausnummer = hausnummer;
        this.plz = plz;
        this.ort = ort;
    }

    public Long getId() {
        return id;
    }

    public String getKvnr() {
        return kvnr;
    }

    public String getName() {
        return name;
    }

    public String getVorname() {
        return vorname;
    }

    public String getStrasse() {
        return strasse;
    }

    public String getHausnummer() {
        return hausnummer;
    }

    public String getPlz() {
        return plz;
    }

    public String getOrt() {
        return ort;
    }
}

File: src/main/kotlin/com/example/dpop/ext_stammdaten/internal/Person.kt
package com.example.dpop.ext_stammdaten.internal

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.LocalDate

@Entity
@Table(name = "person")
class Person(
    @Column(name = "kvnr", unique = true, nullable = false, length = 20)
    var kvnr: String? = null,
    var name: String? = null,
    var vorname: String? = null,
    var strasse: String? = null,
    var hausnummer: String? = null,
    var plz: String? = null,
    var ort: String? = null,
    var geburtsdatum: LocalDate? = null
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
}
